src/Brancher/BrancherDataValidator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hypernode\Deploy\Brancher;
+
+use Hypernode\Deploy\Exception\ValidationException;
+
+class BrancherDataValidator
+{
+    /**
+     * @throws ValidationException
+     */
+    public static function validateLabels(array $labels)
+    {
+        $violations = array_map(
+            fn (string $label) => !preg_match('/^[a-zA-Z0-9-_=]+$/', $label),
+            $labels
+        );
```

Comment:
Adding client-side validation next to the server-side validation might cause some issues. I don't think doing this is a good idea

Reply:
Removed client side validation in 8f62dac